controllers/compaction/metrics.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2023 SAP SE or an SAP affiliate company
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package compaction
+
+import (
+	druidmetrics "github.com/gardener/etcd-druid/pkg/metrics"
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+const (
+	namespaceEtcdDruid  = "etcddruid"
+	subsystemCompaction = "compaction"
+)
+
+var (
+	// metricJobsTotal is the metric used to count the total number of compaction jobs initiated by compaction controller.
+	metricJobsTotal = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: namespaceEtcdDruid,
+			Subsystem: subsystemCompaction,
+			Name:      "jobs_total",
+			Help:      "Total number of compaction jobs initiated by compaction controller.",
+		},
+		[]string{druidmetrics.LabelSucceeded},
+	)
+
+	// metricJobsCurrent is the metric used to expose currently running comapction job. This metric is important to get a sense of total number of compaction job running in a seed cluster..
+	metricJobsCurrent = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespaceEtcdDruid,
+			Subsystem: subsystemCompaction,
+			Name:      "jobs_current",
+			Help:      "Number of currently running comapction job.",
+		},
+		[]string{druidmetrics.EtcdNamespace},
+	)
+
+	// metricJobDurationSeconds is the metric used to expose the time taken to finish running a compaction job.
+	metricJobDurationSeconds = prometheus.NewGaugeVec(
```

Comment:
@ishan16696 , Histogram needs to observe a parameter. So, if a background function runs after a request has been made, we could use histogram to measure how long the background function took to serve the request. But here, we can't observe the job duration time. We need to get out of the controller once the job starts. Only way to measure job duration is by measuring the start time and completion time in job status. That's why we can only set the duration value, not observe the duration. So, we only can use gauge, not histogram.

Reply:
ok, but can this possible ?
```go
metricJobDurationSeconds.With(prometheus.Labels{druidmetrics.LabelSucceeded: druidmetrics.ValueSucceededFalse}). Observe(time.Since(job.Status.StartTime.Time).Seconds())
```
as I'm also seeing this in backup-restore as well, check this
https://github.com/gardener/etcd-backup-restore/blob/d4e6e462ea5db0bda501f966bcd1308349683e50/pkg/initializer/initializer.go#L85

Reply:
As discussed, we are already calculating the job duration from the job status. We can't use Observe function from histogram to calculate job duration because the reconcile doesn't actively track the job. @shreyas-s-rao Would you like to share your thoughts on this?

Reply:
The `Observe` function simply gives Prometheus clients the ability to "record" a value for a given metric. It has no bearing on what kind of metric is being recorded, since both Histogram and Gauge metrics can be set to a specific value using `Observe` and `Set` methods respectively.

Coming to `metricJobDurationSeconds`, I have no string personal opinions on the type of metric to use, but according to what Prometheus recommends, Ishan is right in saying that it should be recorded using a Histogram rather than a Gauge, since durations are generally recorded by Histograms. @ishan16696 thanks for pointing this out.

Reply:
Made the changes